62_CAD/shaders/main_pipeline/fragment_shader.hlsl

```diff
@@ -1,655 +1,602 @@
+            
+            // TODO: we need to emulate dilation and do sdf of neighbouring cells as well. because contours, outlines and shading can bleed into other cells for AA.
+            // [NOTE] Do dilation as last step, when everything else works fine
+
+            DTMSettings dtmSettings = loadDTMSettings(mainObj.dtmSettingsIdx);
+
+            float2 pos = input.getGridDTMScreenSpacePosition();
+            float2 uv = input.getImageUV();
+            const uint32_t textureId = input.getGridDTMHeightTextureID();
+
+            // grid consists of square cells and cells are divided into two triangles:
+            // depending on mode it is
+            // either:        or:
+            // v2a-------v1   v0-------v2b
+            // |  A     / |   | \     B  |
+            // |     /    |   |    \     |
+            // |  /  B    |   |   A   \  |
+            // v0-------v2b   v2a-------v1
+            // 
+
+            // calculate screen space coordinates of vertices of t
+            // he current tiranlge within the grid
+            float3 v[3];
+            nbl::hlsl::shapes::Line<float> outlineLineSegments[2];
+            float outlinePhaseShift;
+            {
+                float2 topLeft = input.getGridDTMScreenSpaceTopLeft();
+                float2 gridExtents = input.getGridDTMScreenSpaceGridExtents();
+                float cellWidth = input.getGridDTMScreenSpaceCellWidth();
+
+                float2 gridSpacePos = uv * gridExtents;
+
+                float2 cellCoords;
+                {
+                    float2 gridSpacePosDivGridCellWidth = gridSpacePos / cellWidth;
+                    cellCoords.x = uint32_t(gridSpacePosDivGridCellWidth.x);
+                    cellCoords.y = uint32_t(gridSpacePosDivGridCellWidth.y);
+                }
+
+                float2 insideCellCoord = gridSpacePos - float2(cellWidth, cellWidth) * cellCoords; // TODO: use fmod instead?
+                
+                const float InvalidHeightValue = asfloat(0x7FC00000);
+                float4 cellHeights = float4(InvalidHeightValue, InvalidHeightValue, InvalidHeightValue, InvalidHeightValue);
+                if (textureId != InvalidTextureIndex)
+                {
+                    const float2 maxCellCoords = float2(round(gridExtents.x / cellWidth), round(gridExtents.y / cellWidth));
+                    const float2 location = (cellCoords + float2(0.5f, 0.5f)) / maxCellCoords;
+
+                    cellHeights = textures[NonUniformResourceIndex(textureId)].Gather(textureSampler, float2(location.x, location.y), 0);
+                }
+
+
+                const E_CELL_DIAGONAL cellDiagonal = resolveGridDTMCellDiagonal(cellHeights);
+                const bool diagonalFromTopLeftToBottomRight = cellDiagonal == E_CELL_DIAGONAL::TOP_LEFT_TO_BOTTOM_RIGHT;
+
+                if (cellDiagonal == E_CELL_DIAGONAL::INVALID)
+                    discard;

```

Comment:
I think you're going to handle it when doing the dilation emulation (when you iterate over all neighbouring cells)
but I'm going to say it again anyways (sorry): we shouldn't discard

Reply:
sure